hack/update-generated-kms-dockerized.sh

```diff
@@ -40,22 +40,22 @@ if [[ -z "$(which protoc)" || "$(protoc --version)" != "libprotoc 3."* ]]; then
 fi
 
 function cleanup {
-	rm -f ${KUBE_KMS_GRPC_ROOT}/service.pb.go.bak
+	rm -f "${KUBE_KMS_GRPC_ROOT}/service.pb.go.bak"
 }
 
 trap cleanup EXIT
 
-gogopath=$(dirname $(kube::util::find-binary "protoc-gen-gogo"))
+gogopath=$(dirname "$(kube::util::find-binary "protoc-gen-gogo")")
 
 PATH="${gogopath}:${PATH}" \
   protoc \
   --proto_path="${KUBE_KMS_GRPC_ROOT}" \
   --proto_path="${KUBE_ROOT}/vendor" \
-  --gogo_out=plugins=grpc:${KUBE_KMS_GRPC_ROOT} ${KUBE_KMS_GRPC_ROOT}/service.proto
+  --gogo_out=plugins=grpc:"${KUBE_KMS_GRPC_ROOT}" "${KUBE_KMS_GRPC_ROOT}/service.proto"
 
 # Update boilerplate for the generated file.
-echo "$(cat hack/boilerplate/boilerplate.generatego.txt ${KUBE_KMS_GRPC_ROOT}/service.pb.go)" > ${KUBE_KMS_GRPC_ROOT}/service.pb.go
+cat hack/boilerplate/boilerplate.generatego.txt "${KUBE_KMS_GRPC_ROOT}/service.pb.go" > tmpfile && mv tmpfile "${KUBE_KMS_GRPC_ROOT}/service.pb.go"
```

Comment:
change reason:
```
In hack/update-generated-kms-dockerized.sh line 57:
echo "$(cat hack/boilerplate/boilerplate.generatego.txt ${KUBE_KMS_GRPC_ROOT}/service.pb.go)" > ${KUBE_KMS_GRPC_ROOT}/service.pb.go
     ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
                                                        ^-- SC2086: Double quote to prevent globbing and word splitting.
                                                                                                ^-- SC2086: Double quote to prevent globbing and word splitting.
```

```
In hack/update-generated-kms-dockerized.sh line 57:
cat hack/boilerplate/boilerplate.generatego.txt "${KUBE_KMS_GRPC_ROOT}/service.pb.go" > "${KUBE_KMS_GRPC_ROOT}/service.pb.go"
                                                ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
                                                                                        ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
```
Ref: https://github.com/koalaman/shellcheck/wiki/SC2094

Reply:
Two concerns with this.. first echo by default appends a `\n` to the end of the file. Can we verify the new pattern does this too?

Additionally, I don't want the tmp file to be left behind if this command gets interrupted. Can we use something like `"${KUBE_KMS_GRPC_ROOT}/service.pb.go.tmp"` and add that filename to the cleanup function on line 42?

Reply:
Thanks for your feedback. @cblecker 
I have verified the new pattern. 
The contents of the files generated by these two methods are the same.
![图片](https://user-images.githubusercontent.com/9354727/56091704-ab3d8180-5ee4-11e9-9458-22698a794f4d.png)